src/server.rs

```diff
@@ -115,6 +116,16 @@ async fn handle_web_request(
                                 .await;
                                 let origin = get_header_string("origin", request.headers());
                                 let user_agent = get_header_string("user-agent", request.headers());
+
+                                // Check domain whitelist if configured
+                                if let Some(whitelist) =
+                                    &settings.authorization.domain_only_whitelist
+                                {
+                                    if !check_domain_whitelist(whitelist, &origin) {
+                                        return;
+                                    }
+                                }
+
```

Comment:
From that I can tell this just checks the domain origin header but this could just be faked by any client as there is no way to verify it. So while it may be somewhat effective against a naive and non targeted spam attack it would not be effective a more targeted or sophisticated attack. 

For development a local relay a local relay maybe more helpful depending on what you're testing and developing.

Reply:
you are right, this is not the right approach. Closing